### PR TITLE
 Add eslint-plugin-jsdoc, improve documentation

### PR DESCRIPTION
**Pull request by fb55**
...and format remaining codebase with prettier. Much easier to review the individual commits.

### CODE REVIEW

File: lib/cheerio.js [hunk 1]
```diff
@@ -25,10 +24,16 @@ var api = [
   require('./api/forms')
 ];
 
-/*
- * Instance of cheerio
+/**
+ * Instance of cheerio. Methods are specified in the modules.
+ * Usage of this constructor is not recommended. Please use $.load instead.
```
[the remark this answers is not in the document]
Comment (fb55) [reply lib/cheerio.js]: That would mean `toArray` isn't documented. Are we okay with that, or should we move it?

File: lib/api/manipulation.js [hunk 2]
```diff
@@ -641,6 +695,8 @@ exports.toString = function() {
  * elements, including their descendants. If `textString` is specified, each
  * selected element's content is replaced by the new text content.
  *
+ * @param {string} str - If specified replacement for the selected element's contents.
```
[the remark this answers is not in the document]
Comment (fb55) [reply lib/api/manipulation.js]: Good catch!

**Comment by fb55** [comment]
All of them are valid, weird that they were not reported. Should be fixed now.

**Comment by jugglinmike** [comment]
Great! I'm ready to merge this. My inclination is to squash into a single commit since rebasing will be a bit tricky at this point. Could you let me know if you'd prefer to keep the history? In that case, I'll use a merge commit.

**Comment by fb55** [comment]
Used squash & merge, no need for the history :)

**Comment by jugglinmike** [comment]
Roger that

**Pull request merged**